### ISSUE TITLE
java.lang.IllegalArgumentException: Missing required <application/> element in .\gen-external-apklibs\in.srain.cube_ultra-ptr_1.0.7\AndroidManifest.xml

### ISSUE THREAD

**Issue by athkalia** [robolectric/robolectric#1612]
An external library does not have an application tag in it's androidmanifest.xml and this is causing robolectric to fail with this:

```
WARNING: no system properties value for ro.build.date.utc
java.lang.IllegalArgumentException: Missing required <application/> element in .\gen-external-apklibs\in.srain.cube_ultra-ptr_1.0.7\AndroidManifest.xml
	at org.robolectric.AndroidManifest.parseAndroidManifest(AndroidManifest.java:155)
	at org.robolectric.AndroidManifest.getPackageName(AndroidManifest.java:474)
	at org.robolectric.AndroidManifest.getResourcePath(AndroidManifest.java:517)
	at org.robolectric.AndroidManifest.getIncludedResourcePaths(AndroidManifest.java:522)
	at org.robolectric.AndroidManifest.getIncludedResourcePaths(AndroidManifest.java:524)
	at org.robolectric.RobolectricTestRunner.createAppResourceLoader(RobolectricTestRunner.java:635)
	at org.robolectric.RobolectricTestRunner.getAppResourceLoader(RobolectricTestRunner.java:627)
	at org.robolectric.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:67)
	at org.robolectric.RobolectricTestRunner.setUpApplicationState(RobolectricTestRunner.java:440)
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:222)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.robolectric.RobolectricTestRunner$1.evaluate(RobolectricTestRunner.java:158)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:74)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:211)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)

java.lang.RuntimeException: java.lang.IllegalArgumentException: Missing required <application/> element in .\gen-external-apklibs\in.srain.cube_ultra-ptr_1.0.7\AndroidManifest.xml
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:226)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.robolectric.RobolectricTestRunner$1.evaluate(RobolectricTestRunner.java:158)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:160)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:74)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:211)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:67)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:134)
Caused by: java.lang.IllegalArgumentException: Missing required <application/> element in .\gen-external-apklibs\in.srain.cube_ultra-ptr_1.0.7\AndroidManifest.xml
	at org.robolectric.AndroidManifest.parseAndroidManifest(AndroidManifest.java:155)
	at org.robolectric.AndroidManifest.getPackageName(AndroidManifest.java:474)
	at org.robolectric.AndroidManifest.getResourcePath(AndroidManifest.java:517)
	at org.robolectric.AndroidManifest.getIncludedResourcePaths(AndroidManifest.java:522)
	at org.robolectric.AndroidManifest.getIncludedResourcePaths(AndroidManifest.java:524)
	at org.robolectric.RobolectricTestRunner.createAppResourceLoader(RobolectricTestRunner.java:635)
	at org.robolectric.RobolectricTestRunner.getAppResourceLoader(RobolectricTestRunner.java:627)
	at org.robolectric.internal.ParallelUniverse.setUpApplicationState(ParallelUniverse.java:67)
	at org.robolectric.RobolectricTestRunner.setUpApplicationState(RobolectricTestRunner.java:440)
	at org.robolectric.RobolectricTestRunner$2.evaluate(RobolectricTestRunner.java:222)
	... 19 more


Process finished with exit code -1

```

If I add an empty application tag on my own 
   
    <application/> 

in that androidmanifest then everything works ok. Can't robolectric assume that if there's no application element then it's just empty?

**Comment by erd** [comment]
This is fixed in 3.0-SNAPSHOT. We're working on getting the release out soon, I promise!

**Comment by athkalia** [comment]
Care to give a rough prediction? Is it like closer to 1 month or half a year  ? :D
